cache.go

```diff
@@ -230,16 +230,23 @@ func (a *Cache[K, V]) SetCapacity(available, max int64) {
 func (a *Cache[K, V]) setCapacity(available, max int64, largerOnly bool) {
 	for {
 		cap := a.cap.Load()
+		size := a.size.Load()
```

Comment:
A bit of color in the description on why this needs to be brought back and such would be very appreciated.

Reply:
Sorry my roast was ready :D

I was testing with our file cache and indeed we need to base available on a delta on top of size, otherwise as soon as we get an available=0, whatever the set capacity can still be filled, whereas ideally it would stop immediately at that size.